metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java

```diff
@@ -130,6 +130,11 @@ protected ScheduledReporter(MetricRegistry registry,
                                 ScheduledExecutorService executor,
                                 boolean shutdownExecutorOnStop,
                                 Set<MetricAttribute> disabledMetricAttributes) {
+
+        if (registry == null) {
+            throw new IllegalArgumentException("MetricRegistry must not be null !!!");
```

Comment:
I'd make this a `throw new NullPointerException("registry == null");` instead

Reply:
Fixed as request